internal/topo/node/watermark_op.go

```diff
@@ -214,7 +217,17 @@ func (w *WatermarkOp) addAndTrigger(ctx api.StreamContext, d *xsql.Tuple) {
 		}
 		// Update watermark
 		if w.sendWatermark {
-			_ = w.Broadcast(&xsql.WatermarkTuple{Timestamp: watermark})
+			if w.delay > 0 {
```

Comment:
This is wrong. Watermark has nothing to do with the processing time (real time). It only deals with the event time. This op is dedicated to send out watermark tuple (tick) and do not need to deal with logics like delay.